build.gradle

```diff
@@ -1719,16 +1718,38 @@ void addMavenPublication(Project project, List<String> projectDependencies) {
     }
 
     compileTargets { t ->
+        // TODO split out the correct 'os' and 'arch' values from 't.name'
+        def os = t.name
+        def arch = "x86"
```

Comment:
This TODO is something that needs to be done in this PR, right? How would this work together with the platform-specific gradle files we have in buildSrc?

Reply:
This should have no influence on how the _platform-specific gradle files we have in buildSrc_ work. This is for determining how the different platform variants are identified by attributes in the metadata. Gradle has predefined attributes and values for `OPERATING_SYSTEM` and `ARCHITECTURE`. 

If my understanding is correct, in a publishing build this code block will be used for each "JavaFX platform". And `t` represents the current platform. What I don't know if how to best map `t` to `OPERATING_SYSTEM` and `ARCHITECTURE`. `t` is of type:

```
class CompileTarget {
    String name;
    String upper;
    String capital;
}
```

So it only has a "name" and IIUC from that name we can somehow get to the  `OPERATING_SYSTEM` and `ARCHITECtURE` – as the "name" is also used to set the classifier of the Jar.

For reference, and clarification, here is the relationship between JavaFXPlatforms and Gradle attribute values in the JavaFX plugin:
https://github.com/openjfx/javafx-gradle-plugin/blob/master/src/main/java/org/openjfx/gradle/JavaFXPlatform.java#L42-L46 

```
    LINUX("linux", "linux-x86_64", OperatingSystemFamily.LINUX, MachineArchitecture.X86_64),
    LINUX_AARCH64("linux-aarch64", "linux-aarch_64", OperatingSystemFamily.LINUX, MachineArchitecture.ARM64),
    WINDOWS("win", "windows-x86_64", OperatingSystemFamily.WINDOWS, MachineArchitecture.X86_64),
    OSX("mac", "osx-x86_64", OperatingSystemFamily.MACOS, MachineArchitecture.X86_64),
    OSX_AARCH64("mac-aarch64", "osx-aarch_64", OperatingSystemFamily.MACOS, MachineArchitecture.ARM64);
```

We can add  a similar list here, but I was wondering if the information is somewhere in the build already (?) And the list above also seems to be incomplete as there are also the _monocle_ variants that are not in there.